lua/autorun/client/cl_nadmodpp.lua

```diff
@@ -20,14 +20,11 @@ net.Receive("nadmod_propowners",function(len)
 	end
 	for i=1, net.ReadUInt(32) do
 		local id, str = net.ReadUInt(16), nameMap[net.ReadUInt(8)]
-		if id==0 then break end
 		if str == "-" then Props[id] = nil PropNames[id] = nil
 		elseif str == "W" then PropNames[id] = "World"
 		elseif str == "O" then PropNames[id] = "Ownerless"
 		else
 			Props[id] = str
-			local ply = player.GetBySteamID(str)
-			PropNames[id] = ply and ply:IsValid() and ply:Nick() or "N/A"
```

Comment:
I think if you keep this around, you'll get a cache of names that'll persist after the original owner has left. Its not perfect, but may as well :)

Or perhaps an
```
if ply and ply:IsValid() then
    PropNames[id] = ply:Nick();
end
```

Reply:
Maybe I should just make the server send the player's name.

Reply:
Could have server keep a separate list of all players previously seen, and then send that once to players on their connect.